app/commands/serial_sequence_command.rb

```diff
@@ -26,11 +26,19 @@ def validate_all_commands
     data.each do |command|
       next if command.valid?
 
-      command.errors.details.each_pair do |attr, command_errors|
-        messages = command.errors.messages[attr]
-        command_errors.each_with_index do |error, idx|
-          errors.add(:"data[#{idx}].#{attr}", messages[idx], error)
-        end
+      validate_command(command)
+    end
+  end
+
+  def validate_command(command)
```

Comment:
This is the simplification mentioned above - its gets the uuid and command from the command itself